### DISCUSSION TITLE
Converting SPA to SSR - Error after compiling because of  IntersectionObserver is not defined ( @ng-web-apis/intersection-observer)

### DISCUSSION THREAD

**Discussion by Rafaelz2516** [taiga-family/taiga-ui#4621] [Q&A] [answered]
Hi, I am currently using Taiga as my default UI library, and it works perfectly in a single-page application (SPA) setup. However, after converting my project to server-side rendering (SSR), I encountered an issue preventing me from starting the server. The error message "IntersectionObserver is not defined" seems to be related to the @ng-web-apis/intersection-observer package, even though this library is not listed in my package.json file.
Did anyone face similar issue if yes? How did you solve it?

![error-1](https://github.com/Tinkoff/taiga-ui/assets/24444427/830cd1fe-51a8-4e0c-92b8-6fadecacbaa5)

Thank you for you help!

**Comment by waterplea** [comment, marked as answer]
You need to provide mocks:
https://taiga-ui.dev/ssr

**Reply by Rafaelz2516** [reply]
Thanks! Sry i missed that part of the documention
